hal/halx86/amd64/x86bios.c

```diff
@@ -216,53 +219,235 @@ x86BiosWriteMemory(
     return STATUS_SUCCESS;
 }
 
-#if 0
+static
+VOID
+FASTCALL 
+x86MemRead(
+    PFAST486_STATE State,
+    ULONG Address,
+    PVOID Buffer,
+    ULONG Size)
+{
+    if (((ULONG64)Address + Size) < 0x100000)
+    {
+        RtlCopyMemory(Buffer, x86BiosMemoryMapping + Address, Size);
+    }
+    else
+    {
+        __debugbreak();
+        RtlFillMemory(Buffer, Size, 0xCC);
+    }
+}
+
+static
+VOID
+FASTCALL
+x86MemWrite(
+    PFAST486_STATE State,
+    ULONG Address,
+    PVOID Buffer,
+    ULONG Size)
+{
+    if (((ULONG64)Address + Size) < 0x100000)
+    {
+        RtlCopyMemory(x86BiosMemoryMapping + Address, Buffer, Size);
+    }
+    else
+    {
+        __debugbreak();
+    }
+}
+
+static
+BOOLEAN
+ValidatePort(
+    USHORT Port,
+    UCHAR Size,
+    BOOLEAN Write)
+{
+    switch (Port)
+    {
+        // VGA: https://wiki.osdev.org/VGA_Hardware#Port_0x3C0
+        case 0x3C0: return (Size == 1) && (Write == TRUE);
```

Comment:
```suggestion
        case 0x3C0: return (Size == 1) && Write;
```
In case of `Write == 2` etc., we shouldn't compare Boolean values.

Reply:
Done